pkg/aggregator/demultiplexer_mock.go

```diff
@@ -74,6 +76,31 @@ func (a *TestAgentDemultiplexer) WaitForSamples(timeout time.Duration) []metrics
 	}
 }
 
+// WaitEventPlatformEvents waits for timeout and eventually returns the event platform events samples received by the demultiplexer.
+func (a *TestAgentDemultiplexer) WaitEventPlatformEvents(eventType string, minEvents int, timeout time.Duration) ([]*message.Message, error) {
+	ticker := time.NewTicker(10 * time.Millisecond)
```

Comment:
Would it make sense to stop the ticker before this function returns?

Reply:
AIUI GC will not collect the internal runtime timer that ticker allocates, only the user facing struct. It's a test so probably doesn't matter much, but I think it's better not to have leaks even in tests, in case this code migrates to prod somehow.

Reply:
> AIUI GC will not collect the internal runtime timer that ticker allocates

@vickenty I think you are right: https://groups.google.com/g/golang-nuts/c/Chx1tCs2QGg

Fixing by adding `defer ticker.Stop()` here https://github.com/DataDog/datadog-agent/pull/10976/commits/22813c1e229145cc61f150b2703eedd3faae9cf0

Also did it for existing `WaitForSamples`